src/editor/actions/actions.js

```diff
@@ -245,58 +245,69 @@ export function moveCursorToNextChunk() {
-    return createNotebookRequest(state.title, state.jsmd, options)
-      .then(notebook => {
-        const message = "Notebook saved to server";
-        dispatch(
-          updateAppMessages({
-            message,
-            messageType: `NOTEBOOK_SAVED`
-          })
-        );
-        dispatch({ type: "ADD_NOTEBOOK_ID", id: notebook.id });
-        window.history.replaceState({}, "", `/notebooks/${notebook.id}`);
-        dispatch({ type: "NOTEBOOK_SAVED" });
-      })
-      .catch(() => {
-        dispatch(
-          updateAppMessages({
-            message: "Error saving notebook.",
-            messageType: "ERROR_SAVING_NOTEBOOK"
-          })
-        );
-      });
+    try {
+      const notebook = await createNotebookRequest(
+        state.title,
+        state.jsmd,
+        options
+      );
+      const message = "Notebook saved to server";
+      dispatch(
+        updateAppMessages({
+          message,
+          messageType: `NOTEBOOK_SAVED`
+        })
+      );
+      dispatch({ type: "ADD_NOTEBOOK_ID", id: notebook.id });
+      window.history.replaceState({}, "", `/notebooks/${notebook.id}`);
+      dispatch({ type: "NOTEBOOK_SAVED" });
+    } catch (err) {
+      dispatch(
+        updateAppMessages({
+          message: "Error saving notebook.",
+          messageType: "ERROR_SAVING_NOTEBOOK"
+        })
+      );
+    }
+
+    return undefined;
   };
 }
 
 export function saveNotebookToServer(appMsg = true) {
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
     const state = getState();
     const notebookId = getNotebookID(state);
     const notebookInServer = Boolean(notebookId);
-    if (notebookInServer) {
-      // Update Exisiting Notebook
-      return updateNotebookRequest(notebookId, state.title, state.jsmd)
-        .then(newRevision => {
-          const message = "Updated Notebook";
-          updateLocalAutosave(state, true);
-          if (appMsg) {
-            dispatch(
-              updateAppMessages({
-                message,
-                messageType: "NOTEBOOK_SAVED"
-              })
-            );
-          }
-          dispatch({ type: "NOTEBOOK_SAVED", newRevisionId: newRevision.id });
-        })
-        .catch(err => {
-          throw Error(err);
-        });
+
+    // Create new notebook
+    if (!notebookInServer) {
+      return createNewNotebookOnServer()(dispatch, getState);
+    }
+    // Update Exisiting Notebook
+    try {
+      const newRevision = await updateNotebookRequest(
+        notebookId,
+        state.title,
+        state.jsmd
+      );
+      const message = "Updated Notebook";
+      updateLocalAutosave(state, true);
+      if (appMsg) {
+        dispatch(
+          updateAppMessages({
+            message,
+            messageType: "NOTEBOOK_SAVED"
+          })
+        );
+      }
+      dispatch({ type: "NOTEBOOK_SAVED", newRevisionId: newRevision.id });
+    } catch (err) {
+      throw Error(err);
```

Comment:
this pattern of
```
catch (err) {
      throw Error(err);
```
is redundant. if we throw anyway, there is no point in catching. this function can be cleaned up further

Reply:
Hmm, looking at this again -- we should probably actually pop up an app message here, unless I'm missing something? I can open up a new pr for that.

Reply:
perhaps. handling the error seems right, but if we don't handle it, we shouldn't rethrow.

also, i think that the try/catch only needs to wrap the actual server request

Reply:
We need the try/catch block around the whole thing, because the try block creates a block scope where the notebook is defined. I suppose we could get around this by defining the notebook object in a let statement, but I don't think it's the right approach-- the whole idea of the try block is to define a bunch of behaviour that should happen in sequence if an operation succeeds.

Reply:
ah, i see. i've always tried to only wrap the thing that i was worried might fail in a `try`, but this makes total sense too. thanks @wlach !